pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -49,7 +49,11 @@ func (cd *clusterDeploy) sync(key string, cluster *v3.Cluster) (runtime.Object,
 		err, updateErr error
 	)
 
-	if key == "" || cluster == nil {
+	if cluster == nil || cluster.DeletionTimestamp != nil {
```

Comment:
earlier we had `key == ""` check here, just want to understand what is the use case and if we still need it

Reply:
Yeah so none of the other controllers check that, and I don't get how the sync handler will get an update for an empty key..